gammapy/image/catalog.py

```diff
@@ -2,18 +2,138 @@
 """Make an image from a source catalog, or simulated catalog, e.g 1FHL 2FGL etc
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
+from collections import OrderedDict
+
 import numpy as np
 from astropy.wcs import WCS
 from astropy.units import Quantity
+from astropy import units as u
 from astropy.table import Table
 
+from .models import Delta2D
+from .core import SkyImage
+from .lists import SkyImageList
+
 
 __all__ = [
+    'CatalogImageEstimator',
     'catalog_image',
     'catalog_table',
 ]
 
 
+BBOX_DELTA2D_PIX = 5
+
+class CatalogImageEstimator(object):
+    """
+    Compute model image for given energy band from a catalog.
+
+    Sources are only filled when their center lies within the image boundaries.
+
+    Parameters
+    ----------
+    reference : `~gammapy.image.SkyImage`
+        Reference sky image.
+    emin : `~astropy.units.Quantity`
+        Lower bound of energy range.
+    emax : `~astropy.units.Quantity`
+        Upper bound of energy range.
+
+    Examples
+    --------
+
+    from astropy import units as u
+    from gammapy.image import SkyImage, CatalogImageEstimator
+    from gammapy.catalog import SourceCatalogGammaCat
+
+    reference = SkyImage.empty(xref=265, yref=-1.5, nxpix=201,
+                               nypix=201, binsz=0.04)
+
+    image_estimator = CatalogImageEstimator(reference=reference,
+                                            emin=1 * u.TeV,
+                                            emax=10 * u.TeV)
+
+    catalog = SourceCatalogGammaCat()
+    result = image_estimator.run(catalog)
+    result['flux'].show()
+
+    """
+
+    def __init__(self, reference, emin, emax):
+        self.reference = reference
+        self.parameters = OrderedDict(emin=emin, emax=emax)
+
+    def flux(self, catalog):
+        """
+        Compute flux image from catalog.
+
+        Sources are only filled when their center lies within the image boundaries.
+
+        Parameters
+        ----------
+        catalog : `~gammapy.catalog.SourceCatalog`
+            Source catalog instance.
+
+        Returns
+        -------
+        image :  `~gammapy.image.SkyImage`
+            Flux sky image.
+        """
+        from ..catalog.gammacat import NoDataAvailableError
+        p = self.parameters
+        image = SkyImage.empty_like(self.reference)
+
+        selection = catalog.select_image_region(image)
+
+        for source in selection:
```

Comment:
Move the code inside the `for source in selection` inside a separate (static?) method, that can be called from elsewhere and tested separately?